crate/crash/command.py

```diff
@@ -580,17 +580,17 @@ def main():
                             args, password=password)
     except (ProgrammingError, LocationParseError) as e:
         msg = getattr(e, 'message', str(e))
-        if '401' in msg and not args.force_passwd_prompt:
-            if is_tty:
-                password = getpass()
+        if '401' in msg and not args.force_passwd_prompt and is_tty:
```

Comment:
I don't know why we would retry creating a shell with the same parameters so I would move the tty check upwards